File: bench/sieve.go
package main

import "fmt"

func sieve(limit int) int {
	isPrime := make([]bool, limit+1)
	for i := 2; i <= limit; i++ {
		isPrime[i] = true
	}

	for i := 2; i*i <= limit; i++ {
		if isPrime[i] {
			for j := i * i; j <= limit; j += i {
				isPrime[j] = false
			}
		}
	}

	count := 0
	for i := 2; i <= limit; i++ {
		if isPrime[i] {
			count++
		}
	}
	return count
}

func main() {
	fmt.Println(sieve(50000000))
}

File: bench/sieve.swift
func sieve(_ limit: Int) -> Int {
    var isPrime = [Bool](repeating: true, count: limit + 1)
    isPrime[0] = false; isPrime[1] = false
    var i = 2
    while i * i <= limit {
        if isPrime[i] {
            var j = i * i
            while j <= limit { isPrime[j] = false; j += i }
        }
        i += 1
    }
    var count = 0
    for i in 2...limit { if isPrime[i] { count += 1 } }
    return count
}
print(sieve(50_000_000))
